libvast/src/system/node.cpp

```diff
@@ -171,6 +172,67 @@ void collect_component_status(node_actor* self,
 
 } // namespace
 
+caf::message dump_command(const invocation& inv, caf::actor_system&) {
+  auto as_yaml = caf::get_or(inv.options, "vast.dump.yaml", false);
+  if (inv.full_name == "dump concepts") {
+    auto self = this_node;
+    auto type_registry = caf::actor_cast<type_registry_actor>(
+      self->state.registry.find_by_label(type_registry_state::name));
+    if (!type_registry)
+      return caf::make_message(
+        make_error(ec::missing_component, type_registry_state::name));
+    caf::error request_error = caf::none;
+    auto rp = self->make_response_promise();
+    self
+      ->request<message_priority::high>(
+        caf::actor_cast<caf::actor>(type_registry),
+        defaults::system::initial_request_timeout, atom::get_v,
+        atom::taxonomies_v)
+      .then(
+        [=](struct taxonomies taxonomies) mutable {
+          auto result = list{};
+          result.reserve(taxonomies.concepts.size());
+          for (auto& [name, definition] : taxonomies.concepts) {
+            auto fields = list{};
+            fields.reserve(definition.fields.size());
+            for (auto& field : definition.fields)
+              fields.push_back(std::move(field));
+            auto concepts = list{};
+            concepts.reserve(definition.concepts.size());
+            for (auto& concept : definition.concepts)
+              fields.push_back(std::move(concept));
+            auto concept = record{
+              {"concept",
+               record{
+                 {"name", std::move(name)},
+                 {"description", std::move(definition.description)},
+                 {"fields", std::move(fields)},
+                 {"concepts", std::move(concepts)},
+               }},
+            };
+            result.push_back(std::move(concept));
+          }
+          if (as_yaml) {
+            auto yaml = to_yaml(data{std::move(result)});
+            if (!yaml) {
+              request_error = std::move(yaml.error());
+              return;
+            }
+            rp.deliver(to_string(std::move(*yaml)));
```

Comment:
```suggestion
            if (auto yaml = to_yaml(data{std::move(result)})
              rp.deliver(to_string(std::move(*yaml)));
            else
              request_error = std::move(yaml.error());
```

Reply:
That can't work, `yaml.error()` cannot be accessed in your version.

Reply:
It's available in the `else` branch just fine. Give it a shot, I do it all the time. 🙂

Reply:
I stand corrected. TIL.